fcrepo-kernel-modeshape/src/test/java/org/fcrepo/kernel/modeshape/services/VersionServiceImplTest.java

```diff
@@ -180,4 +182,23 @@ public void testMixinCreationWhenExplicitlyVersioning() throws RepositoryExcepti
         verify(unversionedNode).addMixin(VERSIONABLE);
     }
 
+    @Test
+    public void testCreateWithInvalidVersionLabels() throws RepositoryException {
+        // [\\s~#@*+%{}<>\\[\\]|\"^]
+        final String[] invalidChars = { " ", "~", "#", "@", "*", "+", "%", "{", "}",
+            "<", ">", "[", "]", "|", "\"", "^" };
+        for (String s : invalidChars) {
+            testLabel(s);
+        }
+    }
+
+    private void testLabel(final String label) {
```

Comment:
Is there no way we can `expect` a specific exception here? Testing for the message in a `Throwable` is often brittle.